pkg/dyninst/procmon/analyze.go

```diff
@@ -131,7 +131,7 @@ func analyzeProcess(
 			"failed to analyze environ", err,
 		)
 	}
-	if ddEnv.serviceName == "" || !ddEnv.diEnabled {
+	if ddEnv.serviceName == "" || ddEnv.serviceVersion == "" || ddEnv.environmentName == "" || !ddEnv.diEnabled {
```

Comment:
Let's make sure we're all on the same page here -- that we need all three of service name, env, and version in order to monitor a process. All of these are indeed needed for that process to be targeted for probing through RC?
From the UI, I seem to be able to create probes without filtering on either env or version, AFAICS.

cc @ajwerner @piob-io

Reply:
yes, all three are required in all other languages, as [described in our docs](https://docs.datadoghq.com/tracing/dynamic_instrumentation/#:~:text=Unified%20Service%20Tagging%20tags%20service%2C%20env%2C%20and%20version%20are%20applied%20to%20your%20deployment.). I think it's safe to require them. Also, in the upcoming 'on by default' efforts, we'd like to target preprod envs which will require anyhow to set the environment.
WDYT?

Reply:
Keeping this conv open until it's fully tested